### PR TITLE
Conform to spec changes at ably/docs#112.

### PR DESCRIPTION
**Pull request by tcard**
This is required by #385. Note that ably/docs#112 is not merged yet.

Split in several commits for ease of consumption.

### CODE REVIEW

**Comment by ricardopereira** [comment]
LGTM, now timeouts are way simple! Great one.

File: Source/ARTRealtime.m [hunk 1]
```diff
@@ -158,20 +139,54 @@ - (void)connect {
...
+        return;
+    case ARTRealtimeConnecting:
+    case ARTRealtimeDisconnected: {
+        [_connection once:^(ARTConnectionStateChange *change) {
+            [self ping:cb];
+        }];
+        return;
+    }
+    case ARTRealtimeConnected:
+        [_pingEventEmitter timed:[_pingEventEmitter once:cb] deadline:10.0 onTimeout:^{
```
Comment (ricardopereira) [inline Source/ARTRealtime.m]: 10s is the default timeout?

File: Source/ARTRealtime.m [hunk 2]
```diff
@@ -238,15 +247,15 @@ - (void)transition:(ARTRealtimeConnectionState)state withErrorInfo:(ARTErrorInfo
...
 
             break;
-        case ARTRealtimeConnected:
-            [self cancelSuspendTimer];
-            break;
-        case ARTRealtimeClosing:
-            [self startCloseTimer];
+        }
+        case ARTRealtimeClosing: {
+            [self unlessStateChangesBefore:10.0 do:^{
```
Comment (ricardopereira) [inline Source/ARTRealtime.m]: Same here.

File: Source/ARTRealtime.m (hunk 1 above)
Comment (tcard) [inline Source/ARTRealtime.m]: It was hardcoded like that before. Fixed at 0aa8cb6.

File: Source/ARTRealtime.m (hunk 2 above)
Comment (tcard) [inline Source/ARTRealtime.m]: Fixed at 0aa8cb6.

**Comment by mattheworiordan** [comment]
FYI, I've reviewed https://github.com/ably/docs/pull/112 and had some comments. it is possible that we may need to revisit this after :(

**Comment by tcard** [comment]
Of course, I counted on that @mattheworiordan, but some parts of this were blocking other things and the refactor was nice anyway. Changing things should be easy now.

**Pull request merged**